src/pocketmine/Player.php

```diff
@@ -3738,6 +3738,7 @@ public function sendPosition(Vector3 $pos, float $yaw = null, float $pitch = nul
 		$pk->headYaw = $yaw;
 		$pk->yaw = $yaw;
 		$pk->mode = $mode;
+		$pk->onGround = $this->onGround;
```

Comment:
I don't know how reliable or safe this is. Players are considered "on ground" when they are within 0.2 blocks of landing vertically. I also don't know if this is used for anything besides teleporting.

Reply:
Walking sounds appear if you send onGround properly.

Reply:
That's not the question here. The question is whether this flag (which is set when the player is within 0.2 blocks of the ground), is accurate enough, and also whether it has any side effects.

Reply:
This fix is ​​safe, no problems were observed